src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -1222,24 +1273,40 @@ def get_lv(lv_name=None, vg_name=None, lv_path=None, lv_uuid=None, lv_tags=None,
     but it can also lead to multiple lvs being found, since a lot of metadata
     is shared between lvs of a distinct OSD.
     """
-    if not any([lv_name, vg_name, lv_path, lv_uuid, lv_tags]):
+    if not any([lv_name, vg_name, lv_path, lv_uuid, lv_tags, pv_name]):
         return None
-    if lvs is None:
+    if (lv_name or vg_name or lv_path or lv_uuid or lv_tags) and pv_name:
+        raise RuntimeError('api.lvm.get_lv: can\'t pass pv_name with other '
+                           'arguments')
+
+    if lvs is None or len(lvs) == 0:
         lvs = Volumes()
-    return lvs.get(
-        lv_name=lv_name, vg_name=vg_name, lv_path=lv_path, lv_uuid=lv_uuid,
-        lv_tags=lv_tags
-    )
+
+    if lv_name or vg_name or lv_path or lv_uuid or lv_tags:
+        return lvs.get(lv_name=lv_name, vg_name=vg_name, lv_path=lv_path,
+                       lv_uuid=lv_uuid, lv_tags=lv_tags)
+
+    if pv_name:
+        filtered_lvs = Volumes(populate=False)
+        output = get_api_lvs(fields="lv_name,devices", unparsed=True, sep=" ")
+        # TODO: move this to process.py, get_api_* methods or wherever is the
```

Comment:
is process.py not trimming the lines?

Reply:
It its, but  I remembering spotting few spaces before disknames; eg. -`['   /dev/sda', '   /dev/sdb']`